engine/sorted_collection/skiplist.cpp

```diff
@@ -268,57 +264,41 @@ Status Skiplist::CheckIndex() {
   return Status::Ok;
 }
 
-bool Skiplist::lockRecordPosition(const DLRecord* record,
-                                  const SpinMutex* record_key_lock,
-                                  std::unique_lock<SpinMutex>* prev_record_lock,
-                                  PMEMAllocator* pmem_allocator,
-                                  HashTable* hash_table) {
+LockTable::GuardType Skiplist::lockRecordPosition(const DLRecord* record,
+                                                  PMEMAllocator* pmem_allocator,
+                                                  HashTable* hash_table,
+                                                  LockTable* lock_table) {
   while (1) {
     DLRecord* prev =
         pmem_allocator->offset2addr_checked<DLRecord>(record->prev);
     DLRecord* next = pmem_allocator->offset2addr<DLRecord>(record->next);
+    PMemOffsetType record_offset = pmem_allocator->addr2offset_checked(record);
     PMemOffsetType prev_offset = pmem_allocator->addr2offset_checked(prev);
     PMemOffsetType next_offset = pmem_allocator->addr2offset_checked(next);
 
-    SpinMutex* prev_lock = hash_table->GetHint(prev->Key()).spin;
-    if (prev_lock != record_key_lock) {
-      if (!prev_lock->try_lock()) {
-        return false;
-      }
-      *prev_record_lock =
-          std::unique_lock<SpinMutex>(*prev_lock, std::adopt_lock);
-    }
+    auto guard = lock_table->MultiGuard({prev_offset, record_offset});
```

Comment:
offset is not a good hash. At least 63 locks per 64 locks in LockTable are wasted.